docs/manual/scala/guide/cluster/code/docs/home/scaladsl/persistence/ShoppingCart.scala

```diff
@@ -0,0 +1,251 @@
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+    // #shopping-cart-create-behavior
+  }
+
+  def createWithTagger(entityContext: EntityContext[Command]): Behavior[Command] = {
+    // #shopping-cart-create-behavior-with-tagger
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      .withTagger(AkkaTaggerAdapter.fromLagom(entityContext, Event.Tag))
+    // #shopping-cart-create-behavior-with-tagger
+  }
+
+  def createSnapshots(entityContext: EntityContext[Command]): Behavior[Command] = {
+    // #shopping-cart-create-behavior-with-snapshots
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      // snapshot every 100 events and keep at most 2 snapshots on db
+      .withRetention(RetentionCriteria.snapshotEvery(numberOfEvents = 100, keepNSnapshots = 2))
+    // #shopping-cart-create-behavior-with-snapshots
+  }
+
+  // #shopping-cart-apply-behavior-creation
+  def apply(entityContext: EntityContext[Command]): Behavior[Command] = {
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      .withTagger(AkkaTaggerAdapter.fromLagom(entityContext, Event.Tag))
+      .withRetention(RetentionCriteria.snapshotEvery(numberOfEvents = 100, keepNSnapshots = 2))
+  }
+  // #shopping-cart-apply-behavior-creation
+
+  implicit val shoppingCartFormat: Format[ShoppingCart] = Json.format
+}
+
+// #shopping-cart-state
+final case class ShoppingCart(
+    items: Map[String, Int],
+    // checkedOutTime defines if cart was checked-out or not:
+    // case None, cart is open
+    // case Some, cart is checked-out
+    checkedOutTime: Option[Instant] = None
+)
+// #shopping-cart-state
+{
+  import ShoppingCart._
+
+  def checkedOut: Boolean = checkedOutTime.nonEmpty
+
+  // #shopping-cart-command-handlers
+  def applyCommand(cmd: Command): ReplyEffect[Event, ShoppingCart] =
+    if (checkedOut) {
```

Comment:
I find it more natural if we first show the commands when the shopping cart is open and then the commands when it is checked out. 

Can we invert it? 
And add a method as:

```scala
def isOpen: Boolean = checkedOutTime.isEmpty
```